core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/TwitterDao.java

```diff
@@ -0,0 +1,122 @@
+package ca.jrvs.apps.twitter.dao;
+
+import ca.jrvs.apps.twitter.dao.helper.HttpHelper;
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.utils.JsonParser;
+import com.google.gdata.util.common.base.PercentEscaper;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class TwitterDao implements CrdDao<Tweet, String> {
+
+    // URI constants
+    private static final String API_BASE_URI = "https://api.twitter.com";
+    private static final String POST_PATH = "/1.1/statuses/update.json";
+    private static final String SHOW_PATH = "/1.1/statuses/show.json";
+    private static final String DELETE_PATH = "/1.1/statuses/destroy/";
+
+    // URI symbols
+    private static final String QUERY_SYM = "?";
+    private static final String AMPERSAND = "&";
+    private static final String EQUAL = "=";
+
+    // Response code
+    private static final int HTTP_OK = 200;
+
+    private HttpHelper httpHelper;
+
+    @Autowired
+    public TwitterDao(HttpHelper httpHelper) {
+        this.httpHelper = httpHelper;
+    }
+
+    @Override
+    public Tweet create(Tweet entity) {
+        try {
+            String uriString = getCreateUri(entity);
+            URI uri = new URI(uriString);
+            HttpResponse response = this.httpHelper.httpPost(uri);
+            return parseResponseBody(response, HTTP_OK);
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
```

Comment:
don't use e.printStackTrace();

you throw an exception on line 47. it's the caller's responsibility to catch or log the exception. no need to print any message here.

Reply:
fix all occurances